packages/terminal/src/node/shell-terminal-server.ts

```diff
@@ -41,4 +43,46 @@ export class ShellTerminalServer extends BaseTerminalServer {
             return Promise.resolve(-1);
         }
     }
+
+    // copied and modified from https://github.com/microsoft/vscode/blob/4636be2b71c87bfb0bfe3c94278b447a5efcc1f1/src/vs/workbench/contrib/debug/node/terminals.ts#L32-L75
```

Comment:
I created a CQ for the following copied code:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21340